### ISSUE TITLE
missing synchronized in VisibleQuestRelay?

### ISSUE THREAD

**Issue by matkoniecz** [westnordost/StreetComplete#1494]
https://github.com/westnordost/StreetComplete/blob/0457f7f4d4e795ade3f302684ea2216c661a940e/app/src/main/java/de/westnordost/streetcomplete/data/VisibleQuestRelay.java#L25

```
if (listener != null) listener.onQuestsCreated(quests, group);
```

is it possible that `setListener` setting listener to null is called between check and and calling `listener.onQuestsCreated`?

It is certainly possible (and far more likely) with modified code used to investigated #1415 

```
		if (listener != null) {
			if(id == 1) {
				Log.wtf(TAG, "Threadsafe relay for VisibleQuestListener - listener was not null, relayed. Relay with id " + id);
			}
			listener.onQuestsCreated(quests, group);
			return true;
		}
```

but it seems to be possible also in a current code.

```
07-26 15:49:44.248 24696-24870/de.westnordost.streetcomplete.debug E/OsmOAuthDialogFragment: Threadsafe relay for VisibleQuestListener - listener was not null, relayed. Relay with id 1
07-26 15:49:44.256 24696-24696/de.westnordost.streetcomplete.debug E/OsmOAuthDialogFragment: listener set to null in VisibleQuestRelay in VisibleQuestRelay with id 1
07-26 15:49:44.274 24696-24870/de.westnordost.streetcomplete.debug E/AndroidRuntime: FATAL EXCEPTION: QuestControllerThread
    Process: de.westnordost.streetcomplete.debug, PID: 24696
    java.lang.NullPointerException: Attempt to invoke interface method 'boolean de.westnordost.streetcomplete.data.VisibleQuestListener.onQuestsCreated(java.util.Collection, de.westnordost.streetcomplete.data.QuestGroup)' on a null object reference
```

There is `public synchronized void setListener` but as I undestand it only prevents multiple setListener happening at once (what by itself is unlikely to be useful).

(opening as a new issue as I am trying to debug main cause of #1415 rather this very rare race condition that may not even exist in the first place)

**Comment by westnordost** [comment]
> is it possible that setListener setting listener to null is called between check and and calling listener.onQuestsCreated?

Yes. it is possible. I wonder why I wrote that it is threadsafe. The class is not threadsafe... Hmm.

**Comment by westnordost** [comment]
addressed in #1500
